pkg/cli/disconnect.go

```diff
@@ -0,0 +1,123 @@
+package cli
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/QMSTR/qmstr/pkg/service"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+var disconnectCmdFlags = struct {
+	edge string
+}{}
+
+var disconnectCmd = &cobra.Command{
+	Use:   "disconnect [type_of_node:attribute:value] [type_of_node:attribute:value]",
+	Short: "Disconnect nodes with specific edges",
+	Long: `Usage: qmstrctl disconnect <that> <this>...
+Disconnect from Node <that> Node(s) <this>.
+	
+input: [type_of_node:attribute:value]
+
+type_of_node:
+	- package
+	- file
+attribute:
+	- hash
+	- path
+	- name
+	- type`,
+	Run: func(cmd *cobra.Command, args []string) {
+		setUpControlService()
+		setUpBuildService()
+		if err := disconnectNodes(cmd, args); err != nil {
+			Log.Fatalf("Disonnect failed: %v", err)
+		}
+		tearDownServer()
+	},
+	Args: cobra.MinimumNArgs(2),
+}
+
+func init() {
+	rootCmd.AddCommand(disconnectCmd)
+	disconnectCmd.Flags().StringVar(&disconnectCmdFlags.edge, "edge", "", "Edge to use when disconnecting nodes")
+}
+
+func disconnectNodes(cmd *cobra.Command, args []string) error {
+	thatID, err := ParseNodeID(args[0])
+	if err != nil {
+		return fmt.Errorf("ParseNodeID fail for %q: %v", args[1], err)
+	}
+	switch thatVal := thatID.(type) {
+	case *service.FileNode:
+		that, err := getUniqueFileNode(thatVal)
+		if err != nil {
+			return fmt.Errorf("get unique \"that\" node fail: %v", err)
+		}
+		err = disconnectFromFileNode(that, args[1:])
+		if err != nil {
+			return fmt.Errorf("disconnect file nodes fail: %v", err)
+		}
+	case *service.PackageNode:
+		_, err := controlServiceClient.GetPackageNode(context.Background(), &service.PackageNode{})
+		if err != nil {
+			return fmt.Errorf("get package node fail: %v", err)
+		}
+	default:
+		return fmt.Errorf("unsuported node type %T", thatVal)
+	}
+	return nil
+}
+
+func disconnectFromFileNode(node *service.FileNode, args []string) error {
+	var deleteNodeMsg *service.DeleteMessage
+	for _, nID := range args {
+		thisID, err := ParseNodeID(nID)
+		if err != nil {
+			return fmt.Errorf("ParseNodeID fail for %q: %v", args[0], err)
+		}
+		switch thisVal := thisID.(type) {
+		// FileNode -> FileNode
+		case *service.FileNode:
+			this, err := getUniqueFileNode(thisVal)
+			if err != nil {
+				return fmt.Errorf("get unique file node fail. please use better matching params: %v", err)
+			}
+			// default edge
+			if disconnectCmdFlags.edge == "" {
+				disconnectCmdFlags.edge = "derivedFrom"
+			}
+			// Which edge
+			switch disconnectCmdFlags.edge {
```

Comment:
switch inside a case is rather bad style. Please clean this up.

Reply:
done